### PR TITLE
[PATCH v4] api: timer: improve documentation

### PR DESCRIPTION
**Pull request by psavol**
Improved timer alloc and restart documentation. Functionality is not changed.

v3: rebased

### CODE REVIEW

File: include/odp/api/spec/timer.h [hunk 1]
```diff
@@ -227,13 +227,20 @@ int odp_timer_pool_info(odp_timer_pool_t timer_pool,
 /**
  * Allocate a timer
  *
- * Create a timer (allocating all necessary resources e.g. timeout event) from
- * the timer pool. The user_ptr is copied to timeouts and can be retrieved
- * using the odp_timeout_user_ptr() call.
+ * Allocates a timer from the timer pool. Depending on timer type, the allocated timer is started
+ * with an odp_timer_start() or odp_timer_periodic_start() call. A timer may be reused multiple
+ * times before freeing it back into the timer pool.
```
Comment (MatiasElo) [inline include/odp/api/spec/timer.h:232]: Perhaps: `Depending on timer type, the allocated timer is started with either odp_timer_start() or odp_timer_periodic_start() call.`

File: include/odp/api/spec/timer.h
Comment (MatiasElo) [inline include/odp/api/spec/timer.h:1]: Copyright can be updated.

File: include/odp/api/spec/timer.h [hunk 2]
```diff
@@ -227,13 +227,20 @@ int odp_timer_pool_info(odp_timer_pool_t timer_pool,
...
+ *
+ * When timer expires, the timeout event defined in timer start parameters (see
+ * odp_timer_start_t::tmo_ev or odp_timer_periodic_start_t::tmo_ev) is sent into the provided
+ * destination queue.
+ *
+ * The provided user pointer value is copied into timeout events when the event type is
+ * ODP_EVENT_TIMEOUT. The value can be retrieved from the event with odp_timeout_user_ptr() call.
```
Comment (MatiasElo) [inline include/odp/api/spec/timer.h:239]: Perhaps: `The value can be retrieved from an event...`

**Review by MatiasElo — APPROVED** [review]
Couple wording suggestions but looking OK.

File: include/odp/api/spec/timer.h (hunk 1 above)
Comment (psavol) [reply include/odp/api/spec/timer.h:232]: Fixed in v2

File: include/odp/api/spec/timer.h
Comment (psavol) [reply include/odp/api/spec/timer.h:1]: Fixed in v2

File: include/odp/api/spec/timer.h (hunk 2 above)
Comment (psavol) [reply include/odp/api/spec/timer.h:239]: Fixed in v2

**Review by jerinjacobk — APPROVED** [review]

**Pull request merged**